### ISSUE TITLE
Please fix dwds-3.1.3 to be compatible to vm_service-4.1.0

### ISSUE THREAD

**Issue by gmpassos** [dart-lang/webdev#1037]
The last version of `vm_service` (4.1.0 / Jun 9, 2020) requires implementation of `VmServiceInterface.getSupportedProtocols`.

`dwds` (3.1.3 [latest] / Jun 2, 2020) is not compatbile with vm_service-4.1.0, since it doesn't implement `ChromeProxyService.getSupportedProtocols`.

Any one after `Jun 9, 2020` that calls `pub global activate webdev` will upgrade to the last version of `vm_service` and won't be able to use `webdev` command.

Please, fix the issue ASAP.

Best regards.

```
Failed to precompile webdev:webdev:
/Users/gmp/.pub-cache/hosted/pub.dartlang.org/dwds-3.1.3/lib/src/services/chrome_proxy_service.dart:40:7: Error: The non-abstract class 'ChromeProxyService' is missing implementations for these members:
 - VmServiceInterface.getSupportedProtocols
Try to either
 - provide an implementation,
 - inherit an implementation from a superclass or mixin,
 - mark the class as abstract, or
 - provide a 'noSuchMethod' implementation.

class ChromeProxyService implements VmServiceInterface {
      ^^^^^^^^^^^^^^^^^^
/Users/gmp/.pub-cache/hosted/pub.dartlang.org/vm_service-4.1.0/lib/src/vm_service.dart:745:24: Context: 'VmServiceInterface.getSupportedProtocols' is defined here.
  Future<ProtocolList> getSupportedProtocols();
                       ^^^^^^^^^^^^^^^^^^^^^
```

**Comment by gmpassos** [comment]
Just to help other developers.

Until `webdev` is fixed, to make it work you need to rollback the active version:

```
$> pub global activate webdev 2.5.6
```

**Comment by DanTup** [comment]
@grouma @bkonyi I hit this in devtools too. I think the change was https://github.com/dart-lang/sdk/commit/54fb6ce3c07719d72eec0a14c00f905b724873c2. I think it's technically non-breaking, but because webdev (and devtools) implement the interface, it breaks them.

Not sure if the best way to prevent this - possibly either considering interface changes here breaking, or webdev/devtools using upper-bounds on constraints to avoid getting newer versions than they were tested with?

**Comment by gmpassos** [comment]
Yes, the change was this one, and `dwds-3.1.3` (released before that, but with dep `vm_service: ^4.0.3`) is not compatible.

The problem is that any one (CI, IDE, new server/pc) that calls `pub global activate webdev` will upgrade to last version of all dependencies and brake `webdev`.

**Comment by gmpassos** [comment]
Created pull request `dart-lang/webdev` https://github.com/dart-lang/webdev/pull/1038 with fix.

**Comment by janvladimirmostert** [comment]
Same issue here
https://stackoverflow.com/questions/62301613/error-the-non-abstract-class-chromeproxyservice-is-missing-implementations-fo
